### ISSUE TITLE
update to continuous actions

### ISSUE THREAD

**Issue by DeepWolf90** [eloialonso/iris#13]
Thank you for sharing the code.

I have been updating the code to work with continuous action environments such as DeepMind Control suite, but it seems like the actor-critic is not learning properly and the actor stuck with a bad policy and not getting better.

Here is what I update in the code:

1) I evaluate my code with cartpole env. with only one continuous action.

2) I discretize the continuous action into 11 bins, so the action space is 11 discrete actions.

3) The reward loss is a mean-squared error as the reward is a real number between 0 and 1.

I have already searched for different values for the learning rate and the cross-entropy coefficient. I also tried to increase the number of bins up to 15.

Any suggestions on how to solve this issue!

**Comment by vmicheli** [comment]
Hey,

The steps you propose make sense. I would add that the [DreamerV2](https://arxiv.org/abs/2010.02193) authors preferred another actor loss for control. Section 2.2 - Actor loss function: “For continuous control, we find dynamics backpropagation to work substantially better”.

Hope that helps!

**Comment by DeepWolf90** [comment]
Thank you for your feedback..

Actually, I discretize the continuous action into 11 bins, so the actor outputs categorical distribution. I had to do this because the transformer-based world model expects action tokens in discrete form. I will test the dynamics backpropagation anyway.

Side Note:-

I noticed that changing the reward loss from cross-entropy loss into MSE loss and predicts a continuous-valued reward instead of categorical reward is the reason for the actor-critic not learning. 

Because when I tested with cartpole-sparse where the reward is sparse and I kept the cross-entropy loss, I can see that the actor-critic is learning.

Is there any explanation for this?!!

**Comment by vmicheli** [comment]
The choice of the actor loss may have more to do with the task you are trying to solve than the way you choose to represent the action space.

Regarding reward prediction: reward classification instead of reward regression seems to work better empirically, for instance see [MuZero](https://arxiv.org/abs/1911.08265) and [DreamerV3](https://arxiv.org/abs/2301.04104).
